src/components/form/Calendar/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { Dispatch, SetStateAction, useEffect, useRef } from 'react';
+import { styled } from '@stitches/react';
+import Calendar from 'react-calendar';
+import 'react-calendar/dist/Calendar.css';
+import dayjs from 'dayjs';
+import ErrorMessage from '../ErrorMessage';
+import { useDisplay } from '@hooks/useDisplay';
+import BottomSheetDialog from '../Select/BottomSheetSelect/BottomSheetDialog';
+interface Props {
+  selectedDate: string | null;
+  setSelectedDate: Dispatch<SetStateAction<string | null>>;
+  isOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+  error?: string;
+}
+
+const CalendarModal = ({ selectedDate, setSelectedDate, isOpen, setIsOpen, error }: Props) => {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const { isDesktop, isMobile, isTablet } = useDisplay();
+
+  const handleOutsideClick = (event: any) => {
+    if (!containerRef.current || !containerRef.current.contains(event.target)) {
+      setIsOpen(false);
+    }
+  };
```

Comment:
여기서 event 의 타입을 click 이벤트로 바꿔줄 수 있을 것 같아요!! 또 아래 useEffect 에서 의존성 배열에 handleOutsideClick 을 넣게 되면, 리렌더링이 될 때마다 같은 함수를 다시 만드는 비효율성이 발생하니 이를 useCallback 을 사용해 방지해주면 좋을 것 같아요!!
```suggestion
    const handleOutsideClick = useCallback((event: any) => {
    if (!containerRef.current || !containerRef.current.contains(event.target)) {
      setIsOpen(false);
    }
  }, []);
```

Reply:
천재 개발자를 보았습니다... 사실 저는 콜백을 사용해서 메모리에 함수를 저장해두는 것에 대한 궁금증이 있읍니다... 메모나 콜백을 모든 곳에서 사용하면 오히려 성능 저하를 일으키니 정말 필요할 때만 메모 or 콜백을 사용하라는 글을 봤었는데요... 저는 그래서
1. 저희 웹이 리렌더링이 많이 발생하는 편이 아니고
2. 해당 함수가 비용이 큰 함수가 아니라
일반 함수로 둬도 괜찮다고 생각을 했어요... 혹시 현수만의 useCallback을 사용하는 기준이 있을까요?

Reply:
useEffect 의 의존성 배열이 수정되면 리렌더링이 발생하게 되는데, 이때 handleOutsideClick 도 이 의존성 배열에 포함되게 되어요!! 만약 리렌더링이 발생하면, handleOutsideClick 은 새로운 참조를 가진 함수로 다시 만들어지게 되고, 그럼 이 영향으로 useEffect 가 다시 실행되고, ... 의 무한 반복이 생겨버려요..!! 따라서 useCallback 을 사용해 함수를 캐싱해주어야 이 무한반복을 피할 수 있어 useCallback 을 사용하는 편이 좋다고 생각했습니다 :)

Reply:
위에 남겨준 코드 리뷰를 반영하여 의존성 배열에 handleOutsideClick을 넣게 되면 -> 해당 함수가 새롭게 만들어지면서 useEffect가 무한으로 실행되기 때문에 useCallback으로 감싸야 한다...!는 뜻과 같은 말일까요...?!! 제가 이해한게 맞나요 ? ㅜ . ㅜ ..